### ISSUE TITLE
TFLite produces incorrect output for reshape-unstack-concat-reshape layout transformation

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#118180]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

3.11

### Current behavior?

TensorFlow Lite produces a different result from Keras for a reshape-unstack-concat-reshape layout transformation.

The model reshapes a [3, 4] tensor to [3, 4, 1], unpacks it along axis=1, concatenates the pieces along axis=0, and reshapes the result back to [3, 4]. Keras produces the expected reordered tensor, while the converted TFLite model returns the original row-major order.

In the reproducer below, Keras returns [10, 50, 90, 20, ...], while TFLite returns [10, 20, 30, 40, ...].

Expected behavior?

The converted TFLite model should match the Keras output for the same reshape, unstack, concat, and reshape operations.

### Standalone code to reproduce the issue

```shell
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "-1"

import numpy as np
import tensorflow as tf

X = np.array(
    [[10., 20., 30., 40.],
     [50., 60., 70., 80.],
     [90., 100., 110., 120.]],
    dtype=np.float32,
)
H, W = X.shape


def numpy_ref(x):
    a = x.reshape(H, W, 1)
    b = np.concatenate(list(np.split(a, W, axis=1)), axis=0)
    return b.reshape(H, W)


class M(tf.keras.Model):
    @tf.function(input_signature=[tf.TensorSpec([H, W], tf.float32)])
    def call(self, x):
        a = tf.reshape(x, [H, W, 1])
        b = tf.concat(tf.unstack(a, axis=1), 0)
        return tf.reshape(b, [H, W])


m = M()
ref = numpy_ref(X)
keras_out = m(tf.constant(X)).numpy()

converter = tf.lite.TFLiteConverter.from_keras_model(m)
tflite_model = converter.convert()

interpreter = tf.lite.Interpreter(
    model_content=tflite_model,
    experimental_op_resolver_type=(
        tf.lite.experimental.OpResolverType.BUILTIN_WITHOUT_DEFAULT_DELEGATES
    ),
)

input_idx = interpreter.get_input_details()[0]["index"]
interpreter.resize_tensor_input(input_idx, [H, W])
interpreter.allocate_tensors()
interpreter.set_tensor(input_idx, X)
interpreter.invoke()

tflite_out = interpreter.get_tensor(interpreter.get_output_details()[0]["index"])

print(f"keras={keras_out.ravel().tolist()}")
print(f"tflite={tflite_out.ravel().tolist()}")

if not np.allclose(tflite_out, ref):
    print("BUG REPRODUCED: TFLite output differs from Keras/reference")
else:
    print("not reproduced")
```

### Relevant log output

```shell
keras=[10.0, 50.0, 90.0, 20.0, 60.0, 100.0, 30.0, 70.0, 110.0, 40.0, 80.0, 120.0]
tflite=[10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0, 90.0, 100.0, 110.0, 120.0]
BUG REPRODUCED: TFLite output differs from Keras/reference
```

**Comment by aravindmurali22** [comment]
Hi @ALinrunrun ,

Thank you for taking the time to report this and for your patience.

TFLite has [transitioned](https://developers.googleblog.com/en/tensorflow-lite-is-now-litert/) to LiteRT, Google's primary runtime for on-device AI. All active development, feature updates, and performance enhancements have shifted  to LiteRT. We encourage you to try running your model with LiteRT. If you still encounter this issue, please open a new issue with the updated details in the [LiteRT Repository](https://github.com/google-ai-edge/LiteRT).

For more information and step-by-step guidance on updating your project, please refer to the [LiteRT Migration Guide](https://developers.google.com/edge/litert/migration).
Thank you again!
